scrapy/exceptions.py

```diff
@@ -11,6 +11,11 @@ class NotConfigured(Exception):
     """Indicates a missing configuration situation"""
     pass
 
+class InvalidValue(TypeError):
```

Comment:
I like the alternative name, but I think it shouldn't be private. This exception will raise if a user-implemented middleware's `process_request` or `process_response` returns an invalid value, it's supposed to appear in the user log.